src/test/java/ru/hogwarts/school/controller/FacultyControllerWebMvcTest.java

```diff
@@ -0,0 +1,234 @@
+package ru.hogwarts.school.controller;
+
+import org.json.JSONObject;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import ru.hogwarts.school.model.Faculty;
+import ru.hogwarts.school.model.Student;
+import ru.hogwarts.school.service.FacultyService;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@WebMvcTest
+@Import(FacultyController.class)
+class FacultyControllerWebMvcTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @SpyBean
```

Comment:
Спай и моки это разные вещи! 
```suggestion
    @MockitoBean
```